docs/kubernetes/kctlr-use-bigip-k8s.rst

```diff
@@ -0,0 +1,188 @@
+.. index::
+   single: BIG-IP; Kubernetes; flannel; Network set-up; VXLAN
+
+.. sidebar:: Docs test matrix
+
+   Documentation manually tested with:
+
+   - BIG-IP v12.1.1
+   - Kubernetes v1.7.5
+   - flannel v0.9.0
+
+.. _bigip-k8s-setup:
+
+Add your BIG-IP device to a Kubernetes Cluster using flannel VXLAN
+==================================================================
+
+This document provides step-by-step instructions for adding a BIG-IP device to a Kubernetes Cluster using flannel VXLAN. For more information about this integration, see :ref:`flannel-bigip-info`.
+
+.. important::
+
+   When you add your BIG-IP device to the Cluster Network, you should run the |kctlr| in ``cluster`` mode (``--pool-member-type=cluster``). See :ref:`cluster mode` for more information.
+
+Tasks
+-----
+
+Complete the following tasks to add a BIG-IP device to a `Kubernetes`_ `Cluster Network`_ using `flannel`_.
+
+===== ==================================================================================
+Step  Task
+===== ==================================================================================
+1.    :ref:`k8s-vxlan-setup`
+----- ----------------------------------------------------------------------------------
+2.    :ref:`k8s-flannel-setup`.
+----- ----------------------------------------------------------------------------------
+3.    :ref:`k8s-bigip-node`
+----- ----------------------------------------------------------------------------------
+4.    :ref:`k8s-assign-ip`
+===== ==================================================================================
+
+\
+
+.. _k8s-vxlan-setup:
+
+Create a VXLAN tunnel on the BIG-IP device
+------------------------------------------
+
+#. Create a new BIG-IP VXLAN profile.
+
+   .. code-block:: console
+
+      admin@BIG-IP(cfg-sync Standalone)(Active)(/Common)(tmos)$ create net tunnels vxlan fl-vxlan port 8472 flooding-type none
+
+#. Verify creation of the VXLAN profile.
+
+   .. code-block:: console
+
+      admin@BIG-IP(cfg-sync Standalone)(Active)(/Common)(tmos)$ list net tunnels vxlan fl-vxlan
+
+
+#. Create a new BIG-IP VXLAN tunnel.
+
+   - Use the BIG-IP VTEP (internal) address as the VXLAN ``local-address``.
+   - Set the ``key`` to ``1`` to grant the BIG-IP device access to all Kubernetes/Flannel projects and subnets.
+
+   .. code-block:: console
+
+      admin@BIG-IP(cfg-sync Standalone)(Active)(/Common)(tmos)$ create net tunnels tunnel flannel_vxlan key 1 profile fl-vxlan local-address 172.16.1.3
+
+#. Verify creation of the VXLAN tunnel.
+
+   .. code-block:: console
+
+      admin@BIG-IP(cfg-sync Standalone)(Active)(/Common)(tmos)$ list net tunnels tunnel flannel_vxlan
+
+
+.. _k8s-flannel-setup:
+
+Deploy flannel for Kubernetes
+-----------------------------
+
+Most users deploy flannel as a `network addon`_ when setting up ``kubeadm``. See Kubernetes` :k8sdocs:`Installing a Pod network <setup/independent/create-cluster-kubeadm/#pod-network>` guide for instructions.
```

Comment:
Again, let's not bring kubeadm into the conversation as there are other methods for installing k8s and certified distributions.